core/src/avm1/function.rs

```diff
@@ -27,15 +28,38 @@ pub struct Avm1Function<'gc> {
     /// The name of the function, if not anonymous.
     name: Option<String>,
 
-    /// The names of the function parameters.
-    params: Vec<String>,
+    /// The number of registers to allocate for this function's private register
+    /// set.
+    ///
+    /// If None, then no register set will be allocated and the preload options
+    /// have no effect.
+    register_count: Option<u8>,
```

Comment:
It's used to distinguish between a function with an empty local register set and one that uses the global register set. From earlier Discord comments regarding registers it sounded like a RegisterCount of 0 would give you an empty register set, rather than access to the global register set. If this isn't the case and `RegisterCount` 0 gives you the global set then we don't need the `Option`.

Reply:
I wasn't sure of the behavior either so I double-checked!

The behavior is this:
If a function tries to access a register < function.register_count, it uses the local register.
If it tries to use register >= function.register_count && register < 4, It uses a global register. (< 4 because there are only 4 global registers. This means it is still possible to change global registers)
Otherwise it's a noop and always returns undefined

I don't think the Flash compiler will ever generate code that depends on the above, but I hand-crafted an SWF to test it. (see below)

So this also means that register_count == 0 is the same as a DefineFunction1 that uses only globals.